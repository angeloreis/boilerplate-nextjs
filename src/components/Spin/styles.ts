import styled from 'styled-components'

export const Wrapper = styled.div`
  border: 2px solid #ccc;
  width: 2rem;
  height: 2rem;
  border-top: 2px solid green;
  border-radius: 50%;
  animation: spin 0.8s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`
