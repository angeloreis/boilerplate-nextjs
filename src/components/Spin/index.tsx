import * as S from './styles'

const Spin = () => <S.Wrapper aria-label="spin-loading" />

export default Spin
