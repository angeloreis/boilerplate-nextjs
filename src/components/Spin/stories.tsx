import { Story, Meta } from '@storybook/react'
import Spin from '.'

export default {
  title: 'Spin',
  component: Spin
} as Meta

export const Default: Story = () => <Spin />
