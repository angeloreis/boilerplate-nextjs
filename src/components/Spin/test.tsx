import { render, screen } from '@testing-library/react'

import Spin from '.'

describe('<Spin />', () => {
  it('should render the heading', () => {
    const { container } = render(<Spin />)

    const spinContainer = screen.getByLabelText('spin-loading')

    expect(spinContainer).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
