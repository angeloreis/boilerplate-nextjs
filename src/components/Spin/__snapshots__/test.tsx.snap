// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spin /> should render the heading 1`] = `
.c0 {
  border: 2px solid #ccc;
  width: 2rem;
  height: 2rem;
  border-top: 2px solid green;
  border-radius: 50%;
  -webkit-animation: spin 0.8s linear infinite;
  animation: spin 0.8s linear infinite;
}

<div
  aria-label="spin-loading"
  class="c0"
/>
`
